code/weapon/weapon.h

```diff
@@ -226,25 +226,30 @@ enum InFlightSoundType
 
 #define MAX_SUBSTITUTION_PATTERNS	10
 
-typedef struct weapon_info {
+class weapon_info
+{
+public:
```

Comment:
Why did you turn the `struct` into a `class` if you make all members public anyway? It would be enough to just remove the typedef to make this a bit more modern.

Reply:
Primarily because I think of classes as having member functions, and structs as not having member functions.  Since weapon_info has these, I thought this would make the definition clearer.

Reply:
For me at least, just having member functions does not mean that is has to be a struct. For me a struct should be used if the type contains a lot of public data members which is the case here.

Reply:
Ok, but we already have `player`, `object`, `ship`, `ship_subsys`, and more - all of which are classes with all-public members.

Reply:
The point I am trying to make here is that you added new clutter to the code that is not necessary. Just because other types exist that also do this weird thing does not mean that it is ok to do so here.

Reply:
I disagree that it is clutter; as I said earlier I thought this would make the definition clearer.  It also makes it consistent with the other places.  But in the interest of getting this approved I'll roll it back.